src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -387,6 +390,23 @@ private static VersionRange GetVersionRange(IVsReferenceItem item)
             return VersionRange.All;
         }
 
+        private static bool GetPropertyBoolOrFalse(
+                IVsReferenceItem item, string propertyName)
+        {
+            try
+            {
+                return MSBuildStringUtility.IsTrue(item.Properties?.Item(propertyName)?.Value);
+            }
+            catch (ArgumentException)
+            {
+            }
+            catch (KeyNotFoundException)
+            {
+            }
```

Comment:
These are possible exceptions from IVsReferenceItem. We want to check if the property exists as true here, all other scenarios should result in false.